### PR TITLE
Support for Custom Binary Paths

### PR DESCRIPTION
**Pull request by bobdenotter**

### CODE REVIEW

File: bolt.php [hunk 1]
```diff
@@ -18,6 +18,34 @@
...
+    if (isset($bin['composer'])) {
+        return $bin['composer'];
+    }
+
+    if (commandExist('composer')) {
+        $composer = locateBinaryPath('composer');
+    }
+
+    if (empty($composer)) {
+        run("cd {{release_path}} && curl -sS https://getcomposer.org/installer | {{bin/php}}");
```
Comment (I-Valchev) [inline bolt.php:42]: Can we add some output here? Would be nice to let the user know we didn't find a composer binary, hence we're fetching one now.

Comment (bobdenotter) [reply bolt.php:42]: This bit is taken straight from Deployer itself. You can't call `__parent()` or somesuch, so i've copied the method and added this bit to allow to override it: 

```
    $bin = get('bin', []);
    if (isset($bin['composer'])) {
        return $bin['composer'];
    }
```

I don't think we should mess with the original parts.

Comment (I-Valchev) [reply bolt.php:42]: Ah, in that case, sure! 👍

**Review by I-Valchev — APPROVED** [review]
👍

**Pull request merged**